external_libraries/CMakeLists.txt

```diff
@@ -105,8 +105,12 @@ if(NOT SYSTEM_YAMLCPP)
   set(YAMLCPP_LIBRARY yaml)
   set(YAMLCPP_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/external_libraries/yaml-cpp/include)
 
-  if(CMAKE_COMPILER_IS_GNUCXX)
-    target_compile_options(yaml PRIVATE -Wno-deprecated-declarations)
+  if(WIN32)
+    if(MSVC) # MSVC needs these exports even for the static library
+      target_compile_definitions(yaml PRIVATE yaml_cpp_EXPORTS YAML_CPP_DLL)
+    else() # for MinGW we need to set YAML_CPP_STATIC_DEFINE as public for some reason
+      target_compile_definitions(yaml PUBLIC YAML_CPP_STATIC_DEFINE)
+    endif()
```

Comment:
I think we might be able to replace all of this stuff with `add_subdirectory(yaml-cpp)` - which will give us a library of `yaml-cpp::yaml-cpp` to link against where we need it.

The surrounding logic here appears to be trying to inject whatever boost library _we_ are using into yaml-cpp - but yaml-cpp have replaced their usage of boost, since 0.7.0 as of: https://github.com/jbeder/yaml-cpp/commit/24fa1b33805c9a91df0f32c46c28e314dd7ad96f - so we can probably simplify this quite a bit:

```cmake
if(NOT SYSTEM_YAMLCPP) # line 95 above!
    message(STATUS "Using bundled yaml-cpp")
    add_subdirectory(yaml-cpp)
endif()
```

and then updating everywhere that tries to link against the one we defined?

Reply:
Thanks for this feedback! I'll try this and will report back.